Add documentation for the new accordion shortcodes

geealbers opened the pull request:
@Erin-Cecele this is the documentation for the new accordion shortcodes, that @hbalenda is close to finishing up. This includes the main `{% accordion %}` shortcode, and then a `{% accordionGlobalControls %}` one that can be used to add "Expand/Collapse All" links wherever they're needed.

Take a look and let me know what you think and if you have any questions. I also had two specific questions for you:

1. Where in the ordering should this page go? I put it after Contributors because it surely seemed less important than that, but then maybe Contributors should also be higher up? Our overall logic of the ordering wasn't clear to me 😬, so feel free to move this new page anywhere you like.
2. Is there anywhere else in the docs where shortcodes are listed or where I should otherwise make a note of these new ones? I think maybe just on the Developers page which we haven't updated yet, but I wasn't sure.

Erin-Cecele commented:
@geealbers let me know if you'd like to discuss my notes above in person. I focused on where non-tech users might get confused by certain terminology. 

As for the location, I would say it should go after "Page Structure". I inherited the ordering as is but always interpreted it, not in terms of importance but in terms of the order of the steps you take to set up a publication. First, you do the outline, then you do each page, then you add images, and references, etc. In that sense, I could see some small tweaks that need to be made to make that more clear (for example, contributors could probably be bumped up.)

I think accordions could/should be added to the Page Structure section. It could also potentially be referenced in Collection Catalogues since this is where I've seen accordions applied the most so far (outside of CASTINGs). 

As a complete side note, should I remove Collapsible Menus for TOC & Sidebar Menu? https://quire.getty.edu/docs-v1/contents-menu/. Or is that coming down the pike?

geealbers commented:
@Erin-Cecele I hate to say it but there aren't any "notes above". Is it possible you entered them all in as part of a review but then didn't submit the review so they didn't get posted?

geealbers commented on content/docs-v1/accordions.md:
```diff
@@ -0,0 +1,140 @@
+
+## Wrap Content in the Accordion Shortcode
+
+The `{% accordion %}` shortcode is a “paired” shortcode and so requires an opening and closing tag, with the text you want hidden in between.
```
Ah yes, I see. How about "...requires an opening and closing tag surrounding the text you want to be hidden."

Erin-Cecele replied:
yes, that works.

geealbers commented on content/docs-v1/accordions.md:
```diff
@@ -0,0 +1,140 @@
+
+## Copy Links to Accordion Sections
+
+When hovering over the summary text and control icon of an accordion section in a web browser, a symbol appears to the left. This is the copy button. Clicking on it will copy the link for the section to your clipboard. 
```
Yes, though I see it'd not clear. I'll just call them headings.

geealbers commented on content/docs-v1/accordions.md:
```diff
@@ -0,0 +1,140 @@
+
+## Customize Accordion Link Names
+
+By default links to accordion sections are based on the summary text parameter, prepended with `#section-`. For example, the link to `{% accordion "## Section One" %}` would be `#section-section-one`.
```
Ah yes, I see it's confusing. Quire doesn't automatically create heading IDs anymore so that approach won't work, but I'll try this section again.

geealbers commented on content/docs-v1/accordions.md:
````diff
@@ -0,0 +1,140 @@
+Note that the parameters are positional (defined by the order they are given), and the "open" parameter need to always be listed third, after the summary text and the id. If your accordion does not have a custom id, make sure the quotes are still there but leave them empty.
+
+```
+{% accordion "## Section One" "" "open" %}
````
Yes, there's probably some work to be done about better defining positional and named parameters, and being consistent about the way we talk about them throughout the docs. Not sure what to do about it in the context of the accordion docs, so am leaving it as a note for the future.

geealbers commented:
@Erin-Cecele thanks for the feedback! A pushed some commits ...

- A second editing pass, specifically to be more clear and consistent around "summary text", "custom id", and the way ids/urls work in general. Along with some misc. fixes
- Added a mention along with a screenshot to the collection catalogue page
- Added it to the list of shortcodes in the Page Content chapter. Once I came across this list, it seemed better to just mention accordions here than to also put it on the Page Structure chapter which is really more about page layouts and project organization rather than what's on the individual pages. And similarly, because the Accordions chapter is a detail of Page Content, I put it after that chapter in the linear ordering.

Take a look when you can, and let me know what you think.

Erin-Cecele commented:
Great. This version is much clearer! And I agree about the placement within Page Content rather than Page Structure. Looks good to go!

geealbers commented:
@Erin-Cecele I just pushed those changes to switch from summary to heading, and to add in a mapping of the parameters. Let me know what you think

Erin-Cecele reviewed (approved):
It might be worth clarifying that the Global Controls shortcode is not paired. Otherwise, this looks good to go!
